pkg/controllers/placement/namespace_status_sync.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2025 The KubeFleet Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	placementv1beta1 "github.com/kubefleet-dev/kubefleet/apis/placement/v1beta1"
+	"github.com/kubefleet-dev/kubefleet/pkg/utils"
+	"github.com/kubefleet-dev/kubefleet/pkg/utils/controller"
+)
+
+// extractNamespaceFromResourceSelectors extracts the namespace name from ResourceSelectors
+// when StatusReportingScope is NamespaceAccessible. Returns empty string if not applicable.
+func extractNamespaceFromResourceSelectors(placementObj placementv1beta1.PlacementObj) string {
+	spec := placementObj.GetPlacementSpec()
+
+	// Only process if StatusReportingScope is NamespaceAccessible.
+	if spec.StatusReportingScope != placementv1beta1.NamespaceAccessible {
+		klog.V(2).InfoS("StatusReportingScope is not NamespaceAccessible", "placement", klog.KObj(placementObj))
+		return ""
+	}
+
+	// CEL validation ensures exactly one Namespace selector exists when NamespaceAccessible.
+	for _, selector := range spec.ResourceSelectors {
+		selectorGVK := schema.GroupVersionKind{
+			Group:   selector.Group,
+			Version: selector.Version,
+			Kind:    selector.Kind,
+		}
+		// Check if this is a namespace selector by comparing with the standard namespace GVK
+		if selectorGVK == utils.NamespaceGVK {
+			return selector.Name
+		}
+	}
+
+	// This should never happen due to CEL validation, but defensive programming.
+	klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("no namespace selector found despite NamespaceAccessible scope")),
+		"Failed to find valid Namespace selector for NamespaceAccessible scope",
+		"placement", klog.KObj(placementObj))
+	return ""
+}
+
+// TODO: Need to handle the case where namespace selector changes on the CRP.
+// syncClusterResourcePlacementStatus creates or updates ClusterResourcePlacementStatus
+// object in the target namespace when StatusReportingScope is NamespaceAccessible.
+func (r *Reconciler) syncClusterResourcePlacementStatus(ctx context.Context, placementObj placementv1beta1.PlacementObj) error {
+	// Only sync for ClusterResourcePlacement objects (not ResourcePlacement).
+	crp, ok := placementObj.(*placementv1beta1.ClusterResourcePlacement)
```

Comment:
We don't allow changing `StatusReportingScope` right, e.g. from `cluster` to `namespaceAccessible`?